From f6b21a0b8d53ec8ab03b0e9fcdd2376c9f2ebd2f Mon Sep 17 00:00:00 2001
From: Alex Tercete <alextercete@gmail.com>
Date: Tue, 6 Jan 2026 15:52:04 +0000
Subject: [PATCH] Support Bazel 9 (#233)

Bazel 9 moves away from the native C/C++ targets, and requires an
implicit dependency on `rules_cc`.

Co-authored-by: Matt Clarkson <matthew.clarkson@arm.com>
---
 MODULE.bazel                        | 1 +
 toolchain/private/cc_toolchains.bzl | 3 ++-
 toolchain/zig_cc_toolchain.bzl      | 6 ++++--
 3 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/MODULE.bazel b/MODULE.bazel
index cfca164f..cbc6933a 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -5,6 +5,7 @@ module(
 
 bazel_dep(name = "bazel_features", version = "1.21.0")
 bazel_dep(name = "platforms", version = "0.0.10")
+bazel_dep(name = "rules_cc", version = "0.2.14")
 
 bazel_dep(name = "rules_go", version = "0.54.0", dev_dependency = True)
 
diff --git a/toolchain/private/cc_toolchains.bzl b/toolchain/private/cc_toolchains.bzl
index 68c4c2ff..efadf184 100644
--- a/toolchain/private/cc_toolchains.bzl
+++ b/toolchain/private/cc_toolchains.bzl
@@ -1,4 +1,5 @@
 load("@hermetic_cc_toolchain//toolchain:zig_cc_toolchain.bzl", "zig_cc_toolchain_config")
+load("@rules_cc//cc/toolchains:cc_toolchain.bzl", "cc_toolchain")
 load(":defs.bzl", "target_structs", "zig_tool_path")
 
 def declare_cc_toolchains(os, zig_sdk_path):
@@ -54,7 +55,7 @@ def declare_cc_toolchains(os, zig_sdk_path):
             visibility = ["//visibility:private"],
         )
 
-        native.cc_toolchain(
+        cc_toolchain(
             name = zigtarget + "_cc",
             toolchain_identifier = zigtarget + "-toolchain",
             toolchain_config = ":%s_cc_config" % zigtarget,
diff --git a/toolchain/zig_cc_toolchain.bzl b/toolchain/zig_cc_toolchain.bzl
index cebe72b6..72fa0c29 100644
--- a/toolchain/zig_cc_toolchain.bzl
+++ b/toolchain/zig_cc_toolchain.bzl
@@ -1,6 +1,6 @@
-load("@bazel_tools//tools/build_defs/cc:action_names.bzl", "ACTION_NAMES")
+load("@rules_cc//cc:action_names.bzl", "ACTION_NAMES")
 load(
-    "@bazel_tools//tools/cpp:cc_toolchain_config_lib.bzl",
+    "@rules_cc//cc:cc_toolchain_config_lib.bzl",
     "artifact_name_pattern",
     "feature",
     "feature_set",
@@ -9,6 +9,8 @@ load(
     "tool",
     "tool_path",
 )
+load("@rules_cc//cc/common:cc_common.bzl", "cc_common")
+load("@rules_cc//cc/toolchains:cc_toolchain_config_info.bzl", "CcToolchainConfigInfo")
 
 all_link_actions = [
     ACTION_NAMES.cpp_link_executable,
